libraries/AP_SensorHead/AP_SensorHead.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include "Protocol.h"
+
+#include <AP_Common/AP_Common.h>
+#include <AP_Baro/AP_Baro.h>
+#include <AP_InertialSensor/AP_InertialSensor.h>
+#include <AP_HAL/AP_HAL.h>
+
+using namespace SensorHead;
+
+/*
+ * Base Class for a Message Handler
+ */
+template <class T>
+class AP_SensorHead_Handler {
+public:
+    virtual void handle(typename T::data_t *data) = 0;
+};
+
+/*
+ * SensorHead Main Class
+ * NOTE: Must be initialized before the Sensors, otherwise registerSensor will
+ * fail.
+ */
+class AP_SensorHead {
+public:
+    // TODO: What is the right number here?
+    /* Receive Thread Rate */
+    static const uint16_t UPDATE_RATE_HZ = 400;
```

Comment:
it depends what we end up able to achieve. I'd like 1kHz if we can do it, but testing will be needed to see if its achievable.